parser/src/main/java/com/linkedin/avroutil1/parser/avsc/AvroParseContext.java

```diff
@@ -71,19 +71,37 @@ public void resolveReferences() {
         //TODO - add context-level issues for dups
 
         //resolve any unresolved references in individual file results from other files
-        externalReferences = new HashMap<>(1);
+        externalReferences = new ArrayList<>();
         for (AvscParseResult singleFile : individualResults) {
-            Map<String, List<SchemaOrRef>> externalRefs = singleFile.getExternalReferences();
-            externalRefs.forEach((fqcn, refs) -> {
-                AvscParseResult otherFile = knownNamedSchemas.get(fqcn);
-                if (otherFile == null) {
+            List<SchemaOrRef> externalRefs = singleFile.getExternalReferences();
+            for (SchemaOrRef ref : externalRefs) {
+                String simpleName = ref.getRef();
+                AvscParseResult simpleNameResolution = knownNamedSchemas.get(simpleName);
+                AvscParseResult inheritedNameResolution = null;
+
+                String inheritedName = ref.getInheritedName();
+                if (inheritedName != null) {
+                    inheritedNameResolution = knownNamedSchemas.get(inheritedName);
+                }
+
+                // The namespace may be inherited from the parent schema's context or may already be defined in the
+                // name. There may be multiple resolutions for a simple name (either from the null namespace or from
+                // the inherited namespace).
+                if (inheritedNameResolution != null) {
+                    ref.setResolvedTo(inheritedNameResolution.getDefinedNamedSchemas().get(inheritedName));
+                    if (simpleNameResolution != null) {
+                        String msg =
+                            "ERROR: Two different schemas found for reference " + simpleName + " with inherited name "
+                                + inheritedName + ". Only one should exist.";
+                        System.err.println(msg);
+                    }
+                } else if (simpleNameResolution != null) {
+                    ref.setResolvedTo(simpleNameResolution.getDefinedNamedSchemas().get(simpleName));
+                } else {
                     //fqcn is unresolved in this context
-                    externalRefs.computeIfAbsent(fqcn, s -> new ArrayList<>(1)).addAll(refs);
-                    return;
+                    externalReferences.add(ref);
```

Comment:
Is there a use in creating an `unresolvedReferences` object?

Reply:
yes - loking them up i other projects/dependencies